nip47/notifications/nip47_notifier.go

```diff
@@ -108,7 +109,17 @@ func (notifier *Nip47Notifier) notifySubscriber(ctx context.Context, app *db.App
 		"appId":        app.ID,
 	}).Debug("Notifying subscriber")
 
-	ss, err := nip04.ComputeSharedSecret(app.NostrPubkey, notifier.keys.GetNostrSecretKey())
+	appWalletPrivKeyBIP32, err := notifier.keys.GetBIP32ChildKey(uint32(app.ID))
```

Comment:
this code is duplicated / slightly different in a few places - maybe we can move it to the `keys` module?

Reply:
Edit: it already is, but maybe we should add a method there to get the hex value directly

Reply:
I think we need to actually check for the app much before to see if it is a child key or not, and if so use the child private key to do the nip04.ComputeSharedSecret for all needed places, which should resolve the duplication.